### DISCUSSION TITLE
delay sometimes after use setOverflow

### DISCUSSION THREAD

**Discussion by EeeLo** [stm32duino/Arduino_Core_STM32#2012] [Libraries] [answered]
I'm using Teensystep stepper lib to control motors, but the start PWM output has some delay sometimes after call setOverflow() frequently, I'm already to set "setPreloadEnable(false)", but seems not working. any good idea about it?
![c011bcac1a5ce21c033439692f29fe2](https://user-images.githubusercontent.com/6959959/234524317-d99008af-2f15-4758-845e-7aae548dd2fb.jpg)

**Comment by fpistm** [comment, marked as answer]
Hi @EeeLo 
Sorry for the delay, looking at the library, it seems only some STMF4xx boards are supported (using HardwareTimer) else a generic timer implementation is used.
https://github.com/luni64/TeensyStep/blob/795d58bcf90401e4d3360265fd7d34ca8beb16ae/src/timer/stm32/TimerField.h#L1

This can explain what you see if you don't use a STM32FAxx based board.
I guess the support for other could be added easily as the HardwareTimer API is common for all stm32.

**Comment by EeeLo** [comment]
Thanks, I already to fix it by never setOverflow() < 10
